core/src/main/java/org/mule/runtime/core/util/collection/EventToMessageSequenceSplittingStrategy.java

```diff
@@ -39,35 +46,56 @@ public EventToMessageSequenceSplittingStrategy(ExpressionSplittingStrategy expre
   @Override
   @SuppressWarnings({"unchecked", "rawtypes"})
   public MessageSequence<?> split(Event event) {
-    Message msg = event.getMessage();
-    Object payload = msg.getPayload().getValue();
-    if (payload instanceof MessageSequence<?>) {
-      return ((MessageSequence<?>) payload);
-    }
-    if (payload instanceof Iterator<?>) {
-      return new IteratorMessageSequence(((Iterator<Object>) payload));
+    if (expressionSplitterStrategy.hasDefaultExpression()) {
```

Comment:
How is the split type exposed via datasense when this logic is in Mule rather than split being done by DW?